### ISSUE TITLE
[Android] - App crashes when using the front camera

### ISSUE THREAD

**Issue by ZXY101** [WebsiteBeaver/react-native-document-scanner-plugin#50]
Hey there, currently the app appears to crash every time I attempt to use the front camera.
Everything works as expected when using the back camera.

### Recreation steps: 
Device: Samsung Galaxy Note10 Lite
Android version: 13

1. Launch document scanner
2. Switch to front cam
3. Take picture
4. Hit `OK` on image preview
5. App crashes

### Error log:
```bash
2023-01-12 04:20:18.149 18800-18800 AndroidRuntime	com.test.mobile	E  FATAL EXCEPTION: main
	Process: com.test.mobile, PID: 18800
	java.lang.RuntimeException: Canvas: trying to draw too large(127844352bytes) bitmap.
		at android.graphics.RecordingCanvas.throwIfCannotDraw(RecordingCanvas.java:266)
		at android.graphics.BaseRecordingCanvas.drawBitmap(BaseRecordingCanvas.java:98)
		at android.graphics.drawable.BitmapDrawable.draw(BitmapDrawable.java:560)
		at android.widget.ImageView.onDraw(ImageView.java:1446)
		at com.websitebeaver.documentscanner.ui.ImageCropView.onDraw(ImageCropView.kt:225)
		at android.view.View.draw(View.java:24395)
		at android.view.View.updateDisplayListIfDirty(View.java:23256)
		at android.view.View.draw(View.java:24125)
		at android.view.ViewGroup.drawChild(ViewGroup.java:4748)
		at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4505)
		at android.view.View.draw(View.java:24398)
		at android.view.View.updateDisplayListIfDirty(View.java:23256)
		at android.view.View.draw(View.java:24125)
		at android.view.ViewGroup.drawChild(ViewGroup.java:4748)
		at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4505)
		at android.view.View.updateDisplayListIfDirty(View.java:23242)
		at android.view.View.draw(View.java:24125)
		at android.view.ViewGroup.drawChild(ViewGroup.java:4748)
		at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4505)
		at android.view.View.updateDisplayListIfDirty(View.java:23242)
		at android.view.View.draw(View.java:24125)
		at android.view.ViewGroup.drawChild(ViewGroup.java:4748)
		at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4505)
		at android.view.View.updateDisplayListIfDirty(View.java:23242)
		at android.view.View.draw(View.java:24125)
		at android.view.ViewGroup.drawChild(ViewGroup.java:4748)
		at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4505)
		at android.view.View.updateDisplayListIfDirty(View.java:23242)
		at android.view.View.draw(View.java:24125)
		at android.view.ViewGroup.drawChild(ViewGroup.java:4748)
		at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4505)
		at com.android.internal.policy.DecorView.dispatchDraw(DecorView.java:1531)
		at android.view.View.draw(View.java:24398)
		at com.android.internal.policy.DecorView.draw(DecorView.java:1504)
		at android.view.View.updateDisplayListIfDirty(View.java:23256)
		at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:753)
		at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:759)
		at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:857)
		at android.view.ViewRootImpl.draw(ViewRootImpl.java:5501)
		at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:5194)
		at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:4356)
		at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2991)
		at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:10665)
		at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1301)
		at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1309)
		at android.view.Choreographer.doCallbacks(Choreographer.java:923)
		at android.view.Choreographer.doFrame(Choreographer.java:852)
		at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1283)
		at android.os.Handler.handleCallback(Handler.java:942)
		at android.os.Handler.dispatchMessage(Handler.java:99)
		at android.os.Looper.loopOnce(Looper.java:226)
		at android.os.Looper.loop(Looper.java:313)
		at android.app.ActivityThread.main(ActivityThread.java:8741)
		at java.lang.reflect.Method.invoke(Native Method)
		at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
		at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

**Comment by ZXY101** [comment]
Hey any feedback on this? Ideally we would never use the front cam in the first place but I'm not sure if there's anyway we can prevent that?

**Comment by dmarcs** [comment]
I couldn't reproduce this on my Pixel

**Comment by ZXY101** [comment]
Hey, looked into this a bit more my side - seems to be the same issue as #57 
It seems like my front camera was taking pictures at a much higher res than the back cam.
My front cam was spitting out images at around `4896x6528` while the back cam was around `2268x4032`.

I adjusted my camera settings down and was able to use the front camera without it crashing.

Looks like you already have a solution in mind, but hope this helps.

**Comment by dmarcs** [comment]
This should be fixed in [0.7.3](https://github.com/WebsiteBeaver/react-native-document-scanner-plugin/tree/0.7.3)

https://github.com/WebsiteBeaver/android-document-scanner/pull/26
